devel/build

```diff
@@ -19,19 +19,33 @@ export GIT_REVISION=$(git describe --tags --abbrev=40 --always --dirty || true)
 # Calling `docker run nextstrain/base` will still only pull down the small base
 # image rather than pulling down the larger nextstrain/base-builder image.
 
-docker build \
+platform=linux/amd64
+
+# `buildx create` is necessary to use a driver that supports multi-platform
+# images.
+builder=$(docker buildx create --driver docker-container)
+
+trap "docker buildx rm $builder" EXIT
+
+docker buildx build \
+    --builder "$builder" \
+    --platform $platform \
     --build-arg CACHE_DATE \
     --build-arg GIT_REVISION \
     --cache-from nextstrain/base-builder \
     --cache-from nextstrain/base \
+    --cache-to type=inline \
```

Comment:
@tsibley Looking at runs from [before](https://github.com/nextstrain/docker-base/commit/9a0f6ae647e422403e3d635518a551999905b012) and [after](https://github.com/nextstrain/docker-base/commit/947a5683e61a128d5a7c1227d75f716da955debb) this was added, I can't make out the benefit of adding `--cache-to type=inline`. The only thing I notice is [the addition of 62.9s](https://github.com/nextstrain/docker-base/runs/7307286281?check_suite_focus=true#step:5:3608):

```
#48 preparing layers for inline cache
#48 DONE 62.9s
```

Reply:
Ah, this is what I was trying to get at in https://github.com/nextstrain/docker-base/pull/49#pullrequestreview-1034754833, but I probably could have been more clear.

I don't think the `--cache-from` will work as we expect without the `--cache-to` I added, as otherwise the caching info won't have been added to the image pushed to the registry and thus can't be imported from the registry later. I haven't tested this, but it's based on my general understanding of how BuildKit works and [this bit of doc about exporting the cache](https://github.com/moby/buildkit#export-cache).

Furthermore, I expect that we won't ever see a local benefit from `--cache-to type=inline` since `--cache-from` always looks at the registry and never the local images (even though we do `--load` a local image with inlined cache). But again, haven't tested this directly.

Reply:
I was confused about the intended effect here, but understand now that `--cache-to` only affects subsequent pulls of the pushed image.

I did some testing:

- Used 62516fef123e95177f2886da3fcc84d225c051d4 to pull from the image built by this PR's source branch.
- Created b330c68f34d2df26e10965f2c28d02825a56fd27 to remove the `--cache-to` and used c1510d47dbea4b5744f42acf5b1917391d2f0f34 to pull from that built image.

At first, I was still confused because both [the run on](https://github.com/nextstrain/docker-base/runs/7326750191?check_suite_focus=true#step:5:53) 62516fef123e95177f2886da3fcc84d225c051d4 and [the run on](https://github.com/nextstrain/docker-base/runs/7327333913?check_suite_focus=true#step:5:48) c1510d47dbea4b5744f42acf5b1917391d2f0f34 show the message `importing cache manifest from ...`, but I think this is due to having `--cache-from`, not necessarily meaning there is any cache available.

But looking closer, it does look like there is a lack of caching when building on an image pushed without `--cache-to`: just go to the logs and search for `RUN pip3 install nextstrain-cli==2.0.0.post1`

Reply:
Thanks for confirming!

Reply:
Hadn't thought of it before, but your testing setup makes me think we should be adding `--cache-from` for the `branch-*` tag when applicable (in addition to the existing `--cache-from` the `latest` tag).

Reply:
Added that idea to #57